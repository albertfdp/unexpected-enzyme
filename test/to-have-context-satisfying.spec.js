import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class User extends Component {
  render() {
    const { name, email } = this.context;

    return (
      <div className="user">
        {name}
        <span className="email">{email}</span>
      </div>
    );
  }
}

User.contextTypes = {
  name: PropTypes.string,
  email: PropTypes.string
};

describe('to-have-context-satisfying', () => {
  it('passes when the passed matches the expected', () => {
    expect(
      mount(<User />, {
        context: { name: 'Harriet Smith', email: 'harriet.smith@example.tld' }
      }),
      'to have context satisfying',
      {
        name: /Smith/,
        email: 'harriet.smith@example.tld'
      }
    );
  });

  it('fails when the actual does not match the expected', () => {
    expect(
      () =>
        expect(
          mount(<User />, {
            context: {
              name: 'Harriet Smith',
              email: 'harriet.smith@example.tld'
            }
          }),
          'to have context satisfying',
          {
            name: /Foo$/,
            email: 'harriet.smith@example.tld'
          }
        ),
      'with error matching snapshot'
    );
  });
});
