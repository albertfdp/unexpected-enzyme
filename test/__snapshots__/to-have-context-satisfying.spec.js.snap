// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-have-context-satisfying fails when the actual does not match the expected 1`] = `
"expected
<User>
  <div className=\\"user\\">
    Harriet Smith
    <span className=\\"email\\">harriet.smith@example.tld</span>
  </div>
</User>
to have context satisfying { name: /Foo$/, email: 'harriet.smith@example.tld' }

{
  name: 'Harriet Smith', // should match /Foo$/
  email: 'harriet.smith@example.tld'
}"
`;
